File: javascript/grains/grains.js
function Grains () {}

Grains.prototype = {
    constructor: Grains,

    square: function (n) { return Math.pow(2, n-1); },

    // The sum of squares is the square of the next int minus 1.
    total: function () { return this.square(65) - 1; }
};

module.exports = Grains;

File: ruby/grains/grains.rb
class Grains
  def self.board
    @board = (1..64).map { |sqr| 2**(sqr-1) }
  end

  def self.square(n)
    board[n-1]
  end

  def self.total
    board.reduce(:+)
  end
end
